keps/sig-node/2681-pod-host-ip/README.md

```diff
@@ -0,0 +1,555 @@
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    HostIPs []HostIP
+```
+
+### PodStatus Internal Representation
+
+PodStatus internally indicates that the original use of HostIP will remain unchanged,
+and HostIPs is added for subsequent use
+
+``` go
+    // HostIP address information for entries in the (plural) HostIPs field.
+    // Each entry includes:
+    //    IP: An IP address allocated to the host.
+    type HostIP struct {
+      // ip is an IP address (IPv4 or IPv6) assigned to the host
+      IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+    }
+
+    // HostIPs holds the IP addresses allocated to the host. If this field is specified, the 0th entry must
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    // +patchStrategy=merge
+    // +patchMergeKey=ip
+    HostIPs []HostIP `json:"hostIPs,omitempty" protobuf:"bytes,14,rep,name=hostIPs" patchStrategy:"merge" patchMergeKey:"ip"`
+```
+
+### Maintaining Compatible Interworking between Old and New Clients
+
+#### V1 to Core (Internal) Conversion
+
+- If only V1 HostIP is provided:
+  - Copy V1 HostIP to core HostIPs[0]
+- Else if only V1 HostIPs[] is provided:
+  - Copy V1 HostIPs[] to core HostIPs[]
+- Else if both V1 HostIP and V1 HostIPs[] are provided:
+  - Verify that V1 HostIP matches V1 HostIPs[0]
+  - Copy V1 HostIPs[] to core HostIPs[]
+- Delete any duplicates in core HostIPs[]
+
+#### Core (Internal) to V1 Conversion
+
+- Copy core HostIPs[0] to V1 HostIP
+- Copy core HostIPs[] to V1 HostIPs[]
+
+
+### Container Environment Variables
+
+The Downward API [status.hostIP](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api)
+will preserve the existing single IP address, and will be set to the default IP
+for each pod. A new environmental variable named `status.hostIPs` will contain a
+space-separated list of IP addresses. The new pod API will have a slice of
+structures for the additional IP addresses. Kubelet will translate the pod
+structures and return hostIPs as a comma-delimited string.
+
+Here is an example of how to define a pluralized `MY_HOST_IPS` environmental
+variable in a pod definition yaml file:
+
+``` yaml
+- name: MY_HOST_IPS
+  valueFrom:
+    fieldRef:
+      fieldPath: status.hostIPs
+```
+
+This definition will cause an environmental variable setting in the pod similar
+to the following:
+
+```
+MY_HOST_IPS=10.20.3.3,fd00:10:20:0:3::3
+```
+
+### Test Plan
+
+Test whether FeatureGate behaves as expected when it is turned on or off
+Test whether Downward API supports `status.hostIPs`
+
+#### Expected behavior
+
+- If PodHostIP FeatureGate is enabled:
+  - If IPv6DualStack FeatureGate is enabled:
+    - `status.hostIPs` there will be both IPv4 and IPv6
+  - Else:
+    - There will only be one
+- Else:
+  - `status.hostIPs` will be null
```

Comment:
and the downward API field isn't recognized; see eg #2055

Reply:
What is written here is the expected result of the test.

Reply:
There is a similar `status.podIPs`, or I got it wrong?

Reply:
I mean, if the `PodHostIPs` feature gate is disabled, then in addition to `status.hostIPs` being null, the downward API field will not be recognized. The apiserver will reject a pod that tries to use `fieldPath: status.hostIPs`.

Reply:
Updated